cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgresSqlStructuredTable.java

```diff
@@ -223,6 +212,71 @@ private StructuredRow resultSetToRow(ResultSet resultSet) throws SQLException {
     }
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> multiScan(Collection<Range> keyRanges,
+                                                    int limit) throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Multiscan ranges {} with limit {}", tableSchema.getTableId(), keyRanges, limit);
+
+    if (keyRanges.isEmpty()) {
+      return CloseableIterator.empty();
+    }
+
+    // Validate all ranges. Also, find if there is any range that is open on both ends.
+    boolean scanAll = false;
+    for (Range range : keyRanges) {
+      fieldValidator.validatePrimaryKeys(range.getBegin(), true);
+      fieldValidator.validatePrimaryKeys(range.getEnd(), true);
+
+      if (range.getBegin().isEmpty() && range.getEnd().isEmpty()) {
+        scanAll = true;
+      }
+    }
+    if (scanAll) {
+      return scan(Range.all(), limit);
+    }
+
+    String query = getMultiScanQuery(keyRanges, limit);
+    try {
+      PreparedStatement statement = connection.prepareStatement(query);
+      statement.setFetchSize(SCAN_FETCH_SIZE);
+
+      int index = 1;
+      for (Range range : keyRanges) {
+        index = setStatementFieldByRange(range, statement, index);
+      }
+      LOG.trace("SQL statement: {}", statement);
+
+      ResultSet resultSet = statement.executeQuery();
+      return new ResultSetIterator(statement, resultSet, tableSchema);
+    } catch (SQLException e) {
+      throw new IOException(String.format("Failed to scan from table %s with ranges %s",
+                                          tableSchema.getTableId().getName(), keyRanges), e);
+    }
+  }
+
+  /**
+   * Generates a SELECT query for scanning over all the provided ranges.
+   *
+   * @param ranges the list of ranges to scan
+   * @param limit number of result
+   * @return a select query
+   */
+  private String getMultiScanQuery(Collection<Range> ranges, int limit) {
```

Comment:
It would be good to add an example of the query generated programmatically through this method in the javadoc. Something similar to this:
https://github.com/cdapio/cdap/blob/release/6.1/cdap-explore/src/main/java/io/cdap/cdap/explore/table/CreateStatementBuilder.java#L37